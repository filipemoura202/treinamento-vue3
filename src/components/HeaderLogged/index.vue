<template>
    <div class="w-full bg-brand-main flex justify-center">
        <div class="w-4/5 h-28 flex items-center justify-between max-w-6xl py-10">
            <div class="w-28 lg:w-36">
                <img src="../../assets/images/logo_white.png" alt="logo" class="src">
            </div>

            <div class="flex md:flex-col">
              <ul class="flex list-none">
                  <li
                    id='credential-button'
                    @click="() => router.push({ name: 'Credentials'})"
                    class="px-6 py-2 font-bold rounded-full text-white focus:outline-none cursor-pointer"
                  >
                    Credenciais
                  </li>
                  <li
                    @click="() => router.push({ name: 'Feedbacks'})"
                    class="px-6 py-2 font-bold rounded-full text-white focus:outline-none cursor-pointer"
                  >
                    Feedbacks
                  </li>
                  <li
                    id='logout-button'
                    @click="handleLogout"
                    class="px-6 py-2 font-bold rounded-full text-white focus:outline-none text-brand-main cursor-pointer bg-white ">
                    {{ labelLogged }}
                  </li>
              </ul>
            </div>
        </div>
    </div>
</template>
<script>
import useStore from '../../hooks/useStore'
import { useRouter } from 'vue-router'
import { cleanCurrentUser } from '../../store/user'
import { computed } from 'vue'
export default {
  setup () {
    const store = useStore('User')
    const router = useRouter()

    const labelLogged = computed(() => {
      if (!store.currentUser.name) {
        return '...'
      }

      return `${store.currentUser.name} (sair)`
    })

    function handleLogout () {
      window.localStorage.removeItem('token')
      cleanCurrentUser()
      router.push({ name: 'Home' })
    }

    return {
      router,
      handleLogout,
      labelLogged
    }
  }

}
</script>
<style>

</style>
