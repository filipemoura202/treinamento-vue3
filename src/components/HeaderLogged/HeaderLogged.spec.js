import { shallowMount } from '@vue/test-utils'
import HeaderLogged from '.'
import { routes } from '../../router'
import { createWebHistory, createRouter } from 'vue-router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

const mockStore = { currentUser: {} }
jest.mock('../../hooks/useStore', () => {
  return () => {
    return mockStore
  }
})

describe('<HeaderLogged />', () => {
  it('Should render Header Logged correctly', async () => {
    router.push('/')
    await router.isReady()

    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [
          router
        ]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should return 3 dots when theres no user logged', async () => {
    router.push('/')
    await router.isReady()

    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [
          router
        ]
      }
    })

    const buttonLogout = wrapper.find('#logout-button')
    expect(buttonLogout.text()).toBe('...')
  })

  it('Should return user name when user logged', async () => {
    router.push('/')
    await router.isReady()
    mockStore.currentUser.name = 'Filipe'
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [
          router
        ]
      }
    })

    const buttonLogout = wrapper.find('#logout-button')
    expect(buttonLogout.text()).toBe('Filipe (sair)')
  })
})
