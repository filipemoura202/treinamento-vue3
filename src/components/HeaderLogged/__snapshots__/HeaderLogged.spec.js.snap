// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> Should render Header Logged correctly 1`] = `
<div class="w-full bg-brand-main flex justify-center">
  <div class="w-4/5 h-28 flex items-center justify-between max-w-6xl py-10">
    <div class="w-28 lg:w-36"><img src="" alt="logo" class="src"></div>
    <div class="flex md:flex-col">
      <ul class="flex list-none">
        <li class="px-6 py-2 font-bold rounded-full text-white focus:outline-none cursor-pointer"> Credenciais </li>
        <li class="px-6 py-2 font-bold rounded-full text-white focus:outline-none cursor-pointer"> Feedbacks </li>
        <li id="logout-button" class="px-6 py-2 font-bold rounded-full text-white focus:outline-none text-brand-main cursor-pointer bg-white">...</li>
      </ul>
    </div>
  </div>
</div>
`;
